app/controllers/api/assignments_controller.rb

```diff
@@ -299,6 +299,24 @@ def submit_file
       upload_file(grouping, only_required_files: assignment.only_required_files)
     end
 
+    def destroy
+      assignment = record
+      if assignment.nil?
```

Comment:
This check isn't necessary, you can remove it. The `check_record` helper is sufficient to check for this